### DISCUSSION TITLE
Dynamic routing in Nuxt 3

### DISCUSSION THREAD

**Discussion by DanielLesko28** [nuxt/nuxt#22953] [Questions]
I am working with dynamic routing in Nuxt 3 i think i am using the NuxtLinks correctly also the path should be alright and also the folder structure where i have pages folder and in it there is users folder and inside it i have [id].vue file but when i click to any user to route to his detail i get just the users page reloaded.
can you help me to fix this issue ?
Thank you

**Comment by renardsas** [comment]
Hi @DanielLesko28,
 We have too little information to assist you.
Would you provide a [reproduction](https://nuxt.com/docs/community/reporting-bugs/#create-a-minimal-reproduction)? 🙏

**Comment by DanielLesko28** [comment]
this is how i am using my nuxt link 

   <NuxtLink :to="`/users/${user.id}`">
                          <Detail />
                        </NuxtLink>

and my flie structure is like this 
pages
  users
    [id].vue
    
when i click on the link i dont get 404 or anything just the page i am currently on reloads
I cannot see why it should not work.

do you need some more info ?

**Comment by renardsas** [comment]
Ok, I've done a small reproduction with /users and /users/id.
[Reproduction](https://stackblitz.com/edit/nuxt3-nuxtpage-pj245f?file=pages%2Fusers%2F%5Bid%5D.vue)
Everything works well. Does this help you? If not, let me know.

**Reply by DanielLesko28** [reply]
I have to reproduce your steps and I will let you know but thank you very much I hope it will work :)

**Reply by DanielLesko28** [reply]
Thank you very much. It really helped me to fix my code. I had wrongly named the files. File where i had the list of users was called users and was not in folder called users. Now I have folder called users where i have index.vue file and [id].vue file and everything works as expected

**Reply by renardsas** [reply]
Ok great, here are some additional information 👇


## Documentation for Routes

You can find the documentation for routes [here](https://nuxt.com/docs/guide/directory-structure/pages).

## User Structure Choices

You have two choices for the user structure:

### 1. Separate Users Page + ID

If you want to have a separate users page and ID, the directory structure would be like this:

```
|-- pages/
  |-- users/
    |-- index.vue
    |-- [id].vue
```

### 2. Nested ID within Users Page

If you want the ID page to be nested within the users page, the directory structure would be:

```
|-- pages/
  |-- users/
    |-- index.vue
  |-- users.vue -> insert <NuxtPage />
```

More details can be found [here](https://nuxt.com/docs/guide/directory-structure/pages#nested-routes) for nested routes.

🚨 For nested routes, there is currently a bug with the use of useAsyncData and useFetch. See https://github.com/nuxt/nuxt/issues/22955
